x-pack/heartbeat/monitors/browser/synthexec/synthexec.go

```diff
@@ -26,8 +30,82 @@ import (
 
 const debugSelector = "synthexec"
 
+// ensure compatability of synthetics by enforcing the installed
+// version never goes beyond this range
+const ExpectedSynthVersion = "<2.0.0"
+
+type PackageJSON struct {
+	Dependencies struct {
+		SynthVersion string `json:"@elastic/synthetics"`
+	} `json:"dependencies"`
+	DevDependencies struct {
+		SynthVersion string `json:"@elastic/synthetics"`
+	} `json:"devDependencies"`
+}
+
+var numberRegex = regexp.MustCompile("[^0-9]+")
+
+// parsed a given dep version by ignoring all range tags (^, = , >, <)
+func parseVersion(version string) string {
+	dotParts := strings.SplitN(version, ".", 4)
+	parsed := []string{}
+
+	for _, v := range dotParts[:3] {
+		value := numberRegex.ReplaceAllString(v, "")
+		parsed = append(parsed, value)
+	}
+	return strings.Join(parsed, ".")
+}
+
+func validateVersion(expected string, current string) error {
+	expectedRange, err := semver.NewConstraint(expected)
+	if err != nil {
+		return err
+	}
+
+	parsed := parseVersion(current)
+	currentVersion, err := semver.NewVersion(parsed)
```

Comment:
One thing to note, this wouldn't work if the user specified `< 2.0.0` as a version. I'm OK with us not covering the semver spec and essentially matching our docs, e.g. `^1.0` etc.

Reply:
We had a discussion and agreed to keep it for now and later improve it if required.